tests/poll.rs

```diff
@@ -473,6 +473,58 @@ fn poll_ok_after_cancelling_pending_ops() {
     handle.join().expect("unable to join thread");
 }
 
+// This test checks the following reregister constraint:
+// The event source must have previously been registered with this instance
+// of `Poll`, otherwise the behavior is undefined.
+//
+// This test is done on Windows and epoll platforms where reregistering a
+// source without a previous register is defined behavior that fail with an
+// error code.
+//
+// On kqueue platforms reregistering w/o registering works but that's not a
+// test goal, so it is not tested.
+#[test]
+#[cfg(any(target_os = "linux", target_os = "windows"))]
+fn reregister_without_register() {
+    let poll = Poll::new().expect("unable to create Poll instance");
+
+    let mut listener = TcpListener::bind(any_local_address()).unwrap();
+
+    assert_eq!(
```

Comment:
Can you use [`assert_error`](https://github.com/tokio-rs/mio/blob/c5f52893ffd4b641d1a457f0197804e69ad666e4/tests/util/mod.rs#L176-L188) to ensure the error string at least mentioned re-registering?